docs/contributing/style-guideline.md

```diff
@@ -105,10 +105,22 @@ All parameters and fields which can be `null` should be annotated with `@Nullabl
 (specifically `org.checkerframework.checker.nullness.qual.Nullable`, which is included by the
 `otel.java-conventions` gradle plugin as a `compileOnly` dependency).
 
-The default is assumed to be non-null, so there is no need to use `@NonNull`.
+There is no need to use `@NonNull`, as this is the default, which should be declared in a
+`package-info.java` file on the root package of each module, e.g.
 
-The `otel.nullaway-conventions` gradle plugin should be used in all modules to perform basic
-nullable usage validation:
+```java
+@DefaultQualifier(value = NonNull.class)
+package io.opentelemetry.instrumentation.api;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.framework.qual.DefaultQualifier;
+```
+
+Public APIs should still defensively check for `null` parameters, even if the parameter is not
```

Comment:
😭 I understand the need, but Idea will be all yellow! :(

Reply:
Ya, that is annoying. I found disabling the inspection below gets rid of that yellow. May not be so bad to disable it once we have enforcement via nullaway.

![image](https://user-images.githubusercontent.com/218610/124803421-f1d88580-df0d-11eb-8517-881ed07cf374.png)